internal/planner/planner.go

```diff
@@ -445,13 +454,20 @@ func (p *Planner) planQuery(q ast.Body, index int, iter planiter) error {
 		return iter()
 	}
 
+	old := p.loc
+	p.loc = q[index].Loc()
+
 	return p.planExpr(q[index], func() error {
-		return p.planQuery(q, index+1, iter)
+		return p.planQuery(q, index+1, func() error {
+			p.loc = old
+			return iter()
+		})
 	})
 }
```

Comment:
I think we need to reset the location before the final return to ensure that the caller's location is not lost. Similarly, the inner-most iterator should do the same.

```golang
old := p.loc
p.loc := q[index].Loc()

err := p.planExpr(q[index], func() error {
   return p.planQuery(q, index+1, func() error {
      curr := p.loc
      p.loc = old
      err := iter()
      p.loc = curr
      return err
   })
})

p.loc = old
return err
```

E.g., when the iterator function on L460 returns back into `planExpr`, the location should be set to the location of that expression. If there was iteration happening inside `planExpr` where the iterator function L460 was called multiple times, I think we could observe this.

Reply:
Thanks! It fixes one test's location that was odd, too. 🎉